### PR TITLE
Query history: Add swagger definitions

### LINKED ISSUE #50305: Query history: Add swagger definitions
Add swagger definitions for query history API

### PR DESCRIPTION
**Pull request by ivanahuckova**
**What this PR does / why we need it**:
This PR:
- Adds swagger definitions
- Adds documentation for migration endpoint
- Adds comments/improves naming

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/grafana/issues/50305

**More context on this**:
Swagger specification is generated using [go-swagger](https://goswagger.io/install.html). This tool uses [annotations](https://goswagger.io/use/spec.html#annotation-syntax) for marking a comment block as a participant for the swagger specification.
The grafana swagger definitions for the OSS endpoints are under [this directory](https://github.com/grafana/grafana/blob/main/pkg/api/docs/definitions) (this directory contain examples of usage of the above annotations that cover the most cases)

For generating the specification, from the grafana root one can run:
`make swagger-api-spec` or `make  swagger-api-spec-mac` (for those using Mac)
 **Note:** that you need to have enterprise enabled so that enterprise routes are included in the generated specification (you need to have `grafana-enterprise` repo in the same folder as `grafana` repo and run /.dev.sh) .

For inspecting the generated specification one can run:
`GF_FEATURE_TOGGLES_ENABLE=swaggerUi make run` and navigate to `http://localhost:3000/swagger-ui`

If the routes are as expected then we can commit the updated `api-spec.json` and `api-merged.json`.

### CODE REVIEW

File: pkg/services/queryhistory/models.go [hunk 1]
```diff
@@ -98,3 +87,26 @@ type QueryHistoryMigrationResponse struct {
...
 }
+
+// CreateQueryInQueryHistoryCommand is the command for adding query history
+// swagger:model
+type CreateQueryInQueryHistoryCommand struct {
+	// UID of the data source for which are queries stored.
+	DatasourceUID string `json:"datasourceUid"`
+	// The JSON model of queries.
+	Queries *simplejson.Json `json:"queries"`
+}
```
Comment (papagian) [inline pkg/services/queryhistory/models.go:98]: Optionally, you can provide an examples values for the fields:
```suggestion
// CreateQueryInQueryHistoryCommand is the command for adding query history
// swagger:model
type CreateQueryInQueryHistoryCommand struct {
	// UID of the data source for which are queries stored.
	// example: PE1C5CBDA0504A6A3
	DatasourceUID string `json:"datasourceUid"`
	// The JSON model of queries.
	// required: true
	// example: [ { "datasourceUid": "PE1C5CBDA0504A6A3", "queries": [ { "refId": "A", "key": "Q-87fed8e3-62ba-4eb2-8d2a-4129979bb4de-0", "scenarioId": "csv_content", "datasource": { "type": "testdata", "uid": "PD8C576611E62080A" } } ], "starred": false, "createdAt": 1643630762, "comment": "debugging" } ] 
	Queries Data `json:"queries"`
}

// swagger:model
type Data *simplejson.Json
```
and then in the swagger editor will appear like:
Screenshot 2022-06-09 at 2 17 40 PM

File: public/api-spec.json [hunk 2]
```diff
@@ -6236,223 +4859,167 @@
...
+            "$ref": "#/responses/internalServerError"
           }
-        ],
+        }
+      },
+      "post": {
+        "description": "Adds new query to query history.",
+        "tags": ["query_history"],
+        "summary": "Add query to query history.",
+        "operationId": "createQuery",
```
Comment (papagian) [inline public/api-spec.json:4884]: body parameter is missing, you need to define as well:
```
// swagger:parameters createQuery
type createQueryParams struct {
	// in:body
	// required:true
	Body  queryhistory.CreateQueryInQueryHistoryCommand `json:"body"`
}
```

File: public/api-spec.json [hunk 3]
```diff
@@ -6236,223 +4859,167 @@
...
-            "in": "path",
-            "required": true
-          }
-        ],
+    "/query-history/migrate": {
+      "post": {
+        "description": "Adds multiple queries to query history.",
+        "tags": ["query_history"],
+        "summary": "Migrate queries to query history.",
+        "operationId": "migrateQueries",
```
Comment (papagian) [inline public/api-spec.json:4906]: Same as above the body parameter is missing

File: public/api-spec.json [hunk 4]
```diff
@@ -6236,223 +4859,167 @@
...
-            "required": true,
-            "schema": {
-              "$ref": "#/definitions/CreateOrUpdateConfigCmd"
-            }
+    "/query-history": {
+      "get": {
+        "description": "Returns a list of queries in the query history that matches the search criteria.\nQuery history search supports pagination. Use the `limit` parameter to control the maximum number of queries returned; the default limit is 100.\nYou can also use the `page` query parameter to fetch queries from any page other than the first one.",
+        "tags": ["query_history"],
+        "summary": "Query history search.",
+        "operationId": "searchQueries",
```
Comment (papagian) [inline public/api-spec.json:4867]: Query parameters are missing; [here](https://github.com/grafana/grafana/blob/a263a0f9bdc1a77b861032c160547aa4651de4e7/pkg/api/docs/definitions/search.go#L22) you can find an example definition

**Review by papagian — COMMENTED** [review]
nice start! I have left some comments

File: docs/sources/developers/http_api/query_history.md [hunk 5]
```diff
@@ -82,7 +82,8 @@ Status codes:
 
 - **200** – OK
```
[the remark this answers is not in the document]
Comment (ivanahuckova) [reply docs/sources/developers/http_api/query_history.md:83]: Added in https://github.com/grafana/grafana/pull/50498/commits/07d169974c0ebd80fe672ae4c1e7ada03ae26a32

File: public/api-spec.json (hunk 4 above)
Comment (ivanahuckova) [reply public/api-spec.json:4867]: Added in https://github.com/grafana/grafana/pull/50498/commits/c77dd362602cf796ceb40d0d8295cb37954e5de0

File: public/api-spec.json [hunk 6]
```diff
@@ -19,45 +19,23 @@
...
-          },
-          "403": {
-            "$ref": "#/responses/forbiddenError"
-          },
-          "500": {
-            "$ref": "#/responses/internalServerError"
-          }
-        }
-      },
+    "/admin/ldap/reload": {
```
[the remark this answers is not in the document]
Comment (ivanahuckova) [reply public/api-spec.json]: Updated in https://github.com/grafana/grafana/pull/50498/commits/763bbda241d646288f1489e5a59f746ef81acd34

**Review by brendamuir — APPROVED** [review]

File: pkg/services/queryhistory/models.go [hunk 7]
```diff
@@ -98,3 +87,30 @@ type QueryHistoryMigrationResponse struct {
...
+// CreateQueryInQueryHistoryCommand is the command for adding query history
+// swagger:model
+type CreateQueryInQueryHistoryCommand struct {
+	// UID of the data source for which are queries stored.
+	// example: PE1C5CBDA0504A6A3
+	DatasourceUID string `json:"datasourceUid"`
+	// The JSON model of queries.
+	// required: true
+	// example: [ { "datasourceUid": "PE1C5CBDA0504A6A3", "queries": [ { "refId": "A", "key": "Q-87fed8e3-62ba-4eb2-8d2a-4129979bb4de-0", "scenarioId": "csv_content", "datasource": { "type": "testdata", "uid": "PD8C576611E62080A" } } ], "starred": false, "createdAt": 1643630762, "comment": "debugging" } ]
+	Queries *simplejson.Json `json:"queries"`
```
Comment (papagian) [inline pkg/services/queryhistory/models.go:100]: the example does not render as expected:
Screenshot 2022-06-10 at 1 19 46 PM
I think you need to define a new type like:
`type Queries *simplejson.Json`
and use this inside the struct (that works for me but I don't know why)

**Review by papagian — APPROVED** [review]

File: pkg/services/queryhistory/models.go (hunk 7 above)
Comment (ivanahuckova) [reply pkg/services/queryhistory/models.go]: Thanks for suggestion! I have updated it in https://github.com/grafana/grafana/pull/50498/commits/b1eb073fbe26fd41351226dde7bdca33838c78f7

Comment (papagian) [reply pkg/services/queryhistory/models.go]: example is rendered properly now 👍

**Pull request merged**